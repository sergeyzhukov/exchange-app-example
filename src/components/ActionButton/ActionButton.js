import React, { PureComponent } from 'react'
import {
  View, Text, StyleSheet, TouchableOpacity,
} from 'react-native'

export default class ActionButton extends PureComponent {
  render() {
    const { title, onPress } = this.props

    return (
      <TouchableOpacity style={styles.container} onPress={onPress}>
        <View style={styles.icon} />
        <Text style={styles.title}>{title}</Text>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    // backgroundColor: 'red',
  },
  icon: {
    width: 24,
    height: 24,
    backgroundColor: 'grey',
  },
  title: {
    marginTop: 12,
    color: '#efefef',
    fontSize: 11,
    fontWeight: '500',
    letterSpacing: -0.2,
    textAlign: 'center',
  },
})
