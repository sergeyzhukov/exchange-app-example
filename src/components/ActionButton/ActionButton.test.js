import React from 'react'
import renderer from 'react-test-renderer'
import ActionButton from './ActionButton'

jest.mock('react-native-vector-icons/Entypo', () => 'Icon')

test('renders correctly', () => {
  const tree = renderer.create(<ActionButton />).toJSON()
  expect(tree).toMatchSnapshot()
})
