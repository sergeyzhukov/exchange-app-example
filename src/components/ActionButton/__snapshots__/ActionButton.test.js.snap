// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
    }
  }
>
  <Icon
    color="white"
    size={24}
  />
  <Text
    style={
      Object {
        "color": "#efefef",
        "fontSize": 11,
        "fontWeight": "500",
        "letterSpacing": -0.2,
        "marginTop": 12,
        "textAlign": "center",
      }
    }
  />
</View>
`;
